Unoptar does not compile

When running `make`, I get the following:

```
% make
gcc -c  -O3 -Wall -Wuninitialized -fomit-frame-pointer -funroll-loops -fstrength-reduce -DNODEBUG `libpng-config --I_opts` -o optar.o optar.c
gcc -c  -O3 -Wall -Wuninitialized -fomit-frame-pointer -funroll-loops -fstrength-reduce -DNODEBUG `libpng-config --I_opts` -o common.o common.c
gcc -c  -O3 -Wall -Wuninitialized -fomit-frame-pointer -funroll-loops -fstrength-reduce -DNODEBUG `libpng-config --I_opts` -o golay.o golay.c
gcc -c  -O3 -Wall -Wuninitialized -fomit-frame-pointer -funroll-loops -fstrength-reduce -DNODEBUG `libpng-config --I_opts` -o parity.o parity.c
gcc -lm -o golay golay.o parity.o
./golay > golay_codes.c
gcc -c  -O3 -Wall -Wuninitialized -fomit-frame-pointer -funroll-loops -fstrength-reduce -DNODEBUG `libpng-config --I_opts` -o golay_codes.o golay_codes.c
gcc -lm -o optar optar.o common.o golay_codes.o parity.o
gcc -c -I/usr/local/include/libpng  -O3 -Wall -Wuninitialized -fomit-frame-pointer -funroll-loops -fstrength-reduce -DNODEBUG `libpng-config --I_opts` -o unoptar.o unoptar.c
gcc -lm -o unoptar -L/usr/local/lib -lpng -lz unoptar.o common.o golay_codes.o parity.o
unoptar.o: In function `get_pixel_interp':
unoptar.c:(.text+0x2a): undefined reference to `floor'
unoptar.c:(.text+0x6e): undefined reference to `floor'
unoptar.o: In function `mark':
unoptar.c:(.text+0x1f3): undefined reference to `floor'
unoptar.c:(.text+0x20e): undefined reference to `floor'
unoptar.o: In function `mark_bad_bit':
unoptar.c:(.text+0x301): undefined reference to `floor'
unoptar.c:(.text+0x3e71): undefined reference to `sqrt'
unoptar.o: In function `dump_newary':
unoptar.c:(.text+0x3ebb): undefined reference to `pow'
unoptar.c:(.text+0x3ede): undefined reference to `floor'
unoptar.c:(.text+0x3f09): undefined reference to `pow'
unoptar.c:(.text+0x3f2c): undefined reference to `floor'
unoptar.c:(.text+0x3f57): undefined reference to `pow'
unoptar.c:(.text+0x3f7a): undefined reference to `floor'
unoptar.c:(.text+0x3fa7): undefined reference to `pow'
unoptar.c:(.text+0x3fca): undefined reference to `floor'
unoptar.o: In function `print_badbit':
unoptar.c:(.text+0x443d): undefined reference to `floor'
unoptar.c:(.text+0x445f): undefined reference to `floor'
unoptar.o: In function `read_png':
unoptar.c:(.text+0x4f0a): undefined reference to `png_create_read_struct'
unoptar.c:(.text+0x4f15): undefined reference to `png_create_info_struct'
unoptar.c:(.text+0x4f27): undefined reference to `png_init_io'
unoptar.c:(.text+0x4f32): undefined reference to `png_read_info'
unoptar.c:(.text+0x4f3d): undefined reference to `png_get_image_width'
unoptar.c:(.text+0x4f4e): undefined reference to `png_get_image_height'
unoptar.c:(.text+0x4fac): undefined reference to `png_get_gAMA'
unoptar.c:(.text+0x4fcb): undefined reference to `png_set_gamma'
unoptar.c:(.text+0x4fd6): undefined reference to `png_get_color_type'
unoptar.c:(.text+0x4fe8): undefined reference to `png_get_bit_depth'
unoptar.c:(.text+0x5026): undefined reference to `png_set_interlace_handling'
unoptar.c:(.text+0x5034): undefined reference to `png_read_update_info'
unoptar.c:(.text+0x5257): undefined reference to `png_read_rows'
unoptar.c:(.text+0x529e): undefined reference to `png_read_rows'
unoptar.c:(.text+0x52b9): undefined reference to `png_read_rows'
unoptar.c:(.text+0x52d4): undefined reference to `png_read_rows'
unoptar.c:(.text+0x52ef): undefined reference to `png_read_rows'
unoptar.o:unoptar.c:(.text+0x530a): more undefined references to `png_read_rows' follow
unoptar.o: In function `read_png':
unoptar.c:(.text+0x534c): undefined reference to `png_set_gamma'
unoptar.c:(.text+0x5366): undefined reference to `png_set_expand'
unoptar.c:(.text+0x537f): undefined reference to `png_set_rgb_to_gray'
unoptar.c:(.text+0x5398): undefined reference to `png_set_strip_alpha'
unoptar.c:(.text+0x53bb): undefined reference to `png_set_rgb_to_gray'
unoptar.c:(.text+0x53db): undefined reference to `png_read_rows'
unoptar.c:(.text+0x53ee): undefined reference to `png_read_rows'
unoptar.c:(.text+0x5401): undefined reference to `png_read_rows'
unoptar.c:(.text+0x5414): undefined reference to `png_read_rows'
unoptar.c:(.text+0x5427): undefined reference to `png_read_rows'
unoptar.o:unoptar.c:(.text+0x543a): more undefined references to `png_read_rows' follow
unoptar.o: In function `read_png':
unoptar.c:(.text+0x5478): undefined reference to `png_read_end'
unoptar.c:(.text+0x54a4): undefined reference to `png_set_expand'
unoptar.c:(.text+0x54b4): undefined reference to `png_set_strip_16'
unoptar.o: In function `main':
unoptar.c:(.text.startup+0x1113): undefined reference to `sqrtf'
unoptar.c:(.text.startup+0x113b): undefined reference to `sqrtf'
unoptar.c:(.text.startup+0x119b): undefined reference to `floor'
unoptar.c:(.text.startup+0x11fd): undefined reference to `floor'
unoptar.c:(.text.startup+0x23c7): undefined reference to `sqrt'
unoptar.c:(.text.startup+0x2432): undefined reference to `sqrt'
unoptar.c:(.text.startup+0x24b3): undefined reference to `asin'
unoptar.c:(.text.startup+0x256a): undefined reference to `asin'
unoptar.c:(.text.startup+0x2615): undefined reference to `asin'
unoptar.c:(.text.startup+0x26b5): undefined reference to `asin'
unoptar.c:(.text.startup+0x274d): undefined reference to `remainder'
unoptar.c:(.text.startup+0x4738): undefined reference to `floor'
unoptar.c:(.text.startup+0x4752): undefined reference to `floor'
unoptar.c:(.text.startup+0x4c90): undefined reference to `floor'
unoptar.c:(.text.startup+0x4d2b): undefined reference to `sqrt'
unoptar.c:(.text.startup+0x4d55): undefined reference to `floorf'
unoptar.c:(.text.startup+0x55ac): undefined reference to `sqrtf'
unoptar.c:(.text.startup+0x55c3): undefined reference to `sqrtf'
unoptar.c:(.text.startup+0x6346): undefined reference to `floor'
unoptar.c:(.text.startup+0x6bc7): undefined reference to `floor'
unoptar.c:(.text.startup+0x6be9): undefined reference to `floor'
unoptar.c:(.text.startup+0x6c19): undefined reference to `floor'
unoptar.c:(.text.startup+0x6f3c): undefined reference to `floor'
unoptar.o:unoptar.c:(.text.startup+0x6f68): more undefined references to `floor' follow
unoptar.o: In function `main':
unoptar.c:(.text.startup+0x83d1): undefined reference to `asin'
unoptar.c:(.text.startup+0x842f): undefined reference to `asin'
unoptar.c:(.text.startup+0x8488): undefined reference to `asin'
unoptar.c:(.text.startup+0x8500): undefined reference to `asin'
collect2: error: ld returned 1 exit status
make: *** [unoptar] Error 1
```

I'm running Ubuntu 12.10, 64-bit.

I found a hint over at http://chaource.livejournal.com/87817.html but had no luck with any permutation of "-lm -lpng".

--- a/Makefile
+++ b/Makefile
@@ -45,4 +45,4 @@ golay: golay.o parity.o
 	gcc $(LDFLAGS) -o $@ $^
 
 unoptar: unoptar.o common.o golay_codes.o parity.o
-	gcc $(LDFLAGS) -o $@ -L/usr/local/lib -lpng -lz $^
+	gcc -o $@ -L/usr/local/lib $^ -lm -lpng -lz

